buildsystem/modules/HandleOpenageCXXOptions.cmake

```diff
@@ -0,0 +1,151 @@
+	endif()
+endmacro()
+
+macro(set_cxx_greater_flags MINIMAL FLAGS)
+	set_cxx_version_flags(${MINIMAL} ${FLAGS} NOT LESS)
+endmacro()
+
+macro(set_cxx_equal_flags MINIMAL FLAGS)
+	set_cxx_version_flags(${MINIMAL} ${FLAGS} "" EQUAL)
+endmacro()
+
+macro(set_cxx_flags FLAGS)
+	set_cxx_version_flags("0.0" ${FLAGS} "" GREATER)
+endmacro()
+
+macro(test_cxx_flag_apply FLAG NAME DONTRUN)
+	if(NOT ${DONTRUN})
+		include(CheckCXXCompilerFlag)
+		check_cxx_compiler_flag("${FLAG}" ${NAME})
+		if(${NAME})
+			set_cxx_flags("${FLAG}")
+		endif()
+	endif()
+endmacro()
+
+macro(set_linker_flags FLAGS)
+	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${FLAGS}")
+	set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${FLAGS}")
+	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${FLAGS}")
+	set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} ${FLAGS}")
+endmacro()
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic")
+
+# check for compiler versions
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+	require_cxx_version("gcc" 4.9)
+	test_cxx_flag_apply("-std=c++14" GCC_SUPPORTS_CPP14 FALSE)
+	set_cxx_greater_flags(4.9 "-fdiagnostics-color=auto")
+	set_cxx_greater_flags(5.0 "-Wsuggest-override")
+
+elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+	require_cxx_version("clang" 3.4)
+	set_cxx_flags("-Wno-mismatched-tags")
+
+	if(APPLE)
+		set_cxx_flags("-stdlib=libc++")
+	endif()
+
+	test_cxx_flag_apply("-std=c++14" CXX_SUPPORTS_CXX14 FALSE)
+	test_cxx_flag_apply("-std=c++1y" CXX_SUPPORTS_CXX1Y CXX_SUPPORTS_CXX14)
+
+	if(NOT (CXX_SUPPORTS_CXX14 OR CXX_SUPPORTS_CXX1Y))
+		message(FATAL_ERROR "compiler doesn't support c++14!")
+	endif()
+
+else() #"Intel", "MSVC", etc..
+	message(WARNING "Using untested compiler, at least I hope it's free software. Continue on your own, warrior.")
+endif()
+
+# If OPENAGE_OPTIMIZATION_LEVEL was not provided, default to auto
+if(NOT OPENAGE_OPTIMIZATION_LEVEL)
+	set(OPENAGE_OPTIMIZATION_LEVEL "auto")
+endif()
+
+if(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "auto")
+	if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+		set(${OPENAGE_OPTIMIZATION_LEVEL} "g")
+	else()
+		set(${OPENAGE_OPTIMIZATION_LEVEL} 2)
+	endif()
+endif()
+
+if(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "0")
+	set_cxx_flags("-O0")
+elseif(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "1")
+	set_cxx_flags("-O1")
+elseif(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "2")
+	set_cxx_flags("-O2")
+elseif(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "g")
+	if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+		set_cxx_flags("-Og")
+	else()
+		set_cxx_flags("-O1")
+	endif()
+elseif(${OPENAGE_OPTIMIZATION_LEVEL} STREQUAL "max")
+	set_cxx_flags("-O3 -march=native")
+
+	if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+		include(ProcessorCount)
+		ProcessorCount(N)
+		if(NOT N EQUAL 0)
+			set_cxx_flags("-flto=${N}")
+			set_linker_flags("-flto=${N}")
+		endif()
+	endif()
+endif()
+
+if(NOT OPENAGE_SANITIZE_MODE)
```

Comment:
That line is checking if `OPENAGE_SANITIZE_MODE` is defined. If it is not defined, i am defaulting it to `none`.